cedar-policy-validator/src/type_error.rs

```diff
@@ -331,9 +339,48 @@ pub(crate) enum UnexpectedTypeHelp {
 }
 
 /// Structure containing details about an incompatible type error.
-#[derive(Debug, Clone, Hash, Eq, PartialEq)]
+#[derive(Diagnostic, Error, Debug, Clone, Hash, Eq, PartialEq)]
+#[diagnostic(help("{context} must have compatible types; {hint}"))]
 pub struct IncompatibleTypes {
     pub(crate) types: BTreeSet<Type>,
+    pub(crate) hint: LubHelp,
+    pub(crate) context: LubContext,
+}
+
+impl Display for IncompatibleTypes {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "The types ")?;
```

Comment:
Assuming this is the displayed error message. We've been using lower case at the start of messages IIRC.
```suggestion
        write!(f, "the types ")?;
```